### PR TITLE
add EUDIW ARF PID spec text example with a yml file

### PR DESCRIPTION
**Pull request by Sakurann**
fixes #210.

update:
- agreed to replace SD-JWT VC example with PID.
- agreed to have only one sd-jwt vc example, so this will also address #144 and close #233.

### CODE REVIEW

**Comment by bc-pi** [comment]
The ["build" is broken](https://github.com/oauth-wg/oauth-selective-disclosure-jwt/actions/runs/5350603690/jobs/9703424502?pr=304) and the editor's draft didn't get generated - https://drafts.oauth.net/oauth-selective-disclosure-jwt/sakurann-pid-ex/draft-ietf-oauth-selective-disclosure-jwt.html#name-key-binding-jwt-2 404s

**Review by bc-pi — DISMISSED** [review]
needs some fixin'

**Comment by Sakurann** [comment]
fixed - thank you, Brian!

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 1]
```diff
@@ -1407,6 +1413,41 @@ After the validation, the Verifier will have the following data for further proc
 
 <{{examples/jsonld/verified_contents.json}}
 
+## Example 5: EUDIW ARF Person Identification Data (PID)  {#person_identification_data}
+
+This example illustrates how to use the artifacts defined in this specification to secure
+Person Identification Data (PID) based on the SD-JWT format as defined in [@EUDIW.ARF].
+
+This example uses the following object as the set of claims that the Issuer is issuing:
```
Comment (danielfett) [inline draft-ietf-oauth-selective-disclosure-jwt.md:1421]:
```suggestion
This example uses the following object as the SD-JWT input claim set:
```

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 2]
```diff
@@ -1407,6 +1413,41 @@ After the validation, the Verifier will have the following data for further proc
...
+The Disclosures of this SD-JWT are as follows:
+
+{{examples/pid/disclosures.md}}
+
+The Issuer added the following decoy digests:
+
+{{examples/pid/decoy_digests.md}}
+
+A presentation of the SD-JWT without a Key Binding JWT that discloses only biometrics based holder binding (`binding` of type `BiometricBinding`)
+nationality, place of birth and the fact that the person is over 18 years old could look as follows:
```
Comment (danielfett) [inline draft-ietf-oauth-selective-disclosure-jwt.md:1442]:
```suggestion
A presentation of the SD-JWT without a Key Binding JWT that discloses only biometrics based holder binding (`binding` of type `BiometricBinding`), 
nationality, place of birth, and the fact that the person is over 18 years old could look as follows:
```

**Review by danielfett — APPROVED** [review]

**Review by tlodderstedt — COMMENTED** [review]
Couple of remarks:

- the binding design deviates from the standard design ("binding" array with typed entries). What is the rationale for that? I personally would prefer to add biometric binding in parallel to the usual "cnf".
- The PID rule book does not defined biometric data yet. I suggest to include this in the example once there is more clarity around that.
- Section A.3. already has a SD-JWT VC example.  I think this example should go into the SD-JWT VC spec as it is another instance of an SD-JWT VC.

**Comment by Sakurann** [comment]
@tlodderstedt Binding design and the usage of biometrics data is based on your original PR 210. Shall we replace 4a example with this one? But if you don’t see the need for this example anymore, I am not going to die on this hill.

File: examples/pid/specification.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+holder_disclosed_claims:
+  binding:
+    - false
+    - true
+  nationalities:
+    - true
+  is_over_18: true
+  place_of_birth: true
+
+add_decoy_claims: true
```
Comment (bc-pi) [inline examples/pid/specification.yml:41]:
```suggestion
add_decoy_claims: false
```

The nested decoy digests (e.g., https://drafts.oauth.net/oauth-selective-disclosure-jwt/sakurann-pid-ex/draft-ietf-oauth-selective-disclosure-jwt.html#appendix-A.5-9) make the disclosure content hard to follow. We might want to look at what the tooling does for nested decoys but in the meantime just not do decoys in this example. I think they only detract.

File: examples/pid/specification.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        y: ZxjiWWbZMQGHVWKVQ4hbSIirsVfuecCE6t4jT9F2HZQ
+  - !sd
+    type: BiometricBinding
+    template: "..."
+  !sd given_name: Erika
+  !sd family_name: Mustermann
+  !sd nationalities: [!sd "DE"]
+  !sd birth_family_name: Schmidt
+  !sd birthdate: '1973-01-01'
+  !sd place_of_birth: Regensburg
```
Comment (peppelinux) [inline examples/pid/specification.yml:22]: [ARF PID Rulebook v.1.0.0] specifies ‘Place of Birth’ as an optional data element in a PID attribute, definfing  the following data elements related to this PID attribute:

∙ birth_place (optional)
∙ birth_country (optional)
∙ birth_state (optional)
∙ birth_city (optional)

or a JSON object like this

````
"place_of_birth":{
    "region":"MI",
    "locality":"F205",
    "country": ...,
    "state": ...,
    "city": ...,
}
````

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 5]
```diff
@@ -1407,6 +1413,41 @@ After the validation, the Verifier will have the following data for further proc
...
+
+The Disclosures of this SD-JWT are as follows:
+
+{{examples/pid/disclosures.md}}
+
+The Issuer added the following decoy digests:
+
+{{examples/pid/decoy_digests.md}}
+
+A presentation of the SD-JWT without a Key Binding JWT that discloses only biometrics based holder binding (`binding` of type `BiometricBinding`)
```
[the remark this answers is not in the document]
Comment (Sakurann) [reply draft-ietf-oauth-selective-disclosure-jwt.md:1441]: ok, changed to disclosing cryptographic key binding :)

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 6]
```diff
@@ -1407,6 +1413,41 @@ After the validation, the Verifier will have the following data for further proc
...
+nationality, place of birth and the fact that the person is over 18 years old could look as follows:
+
+<{{examples/pid/sd_jwt_presentation.txt}}
+
+After the verification of the data, the Verifier will
+pass the following result on to the application for further processing:
```
[the remark this answers is not in the document]
Comment (Sakurann) [reply draft-ietf-oauth-selective-disclosure-jwt.md:1447]: i think it assumes that the verifier has two components - the part that parses/validates sd-jwt and the part that uses the validated data - the latter is referred to as application i think...

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 7]
```diff
@@ -1407,6 +1413,41 @@ After the validation, the Verifier will have the following data for further proc
...
+
+<{{examples/pid/verified_contents.json}}
+
 ## Elliptic Curve Key Used in the Examples
 
 The following Elliptic Curve public key, represented in JWK format, can be used to validate the Issuer signatures in the above examples:
```
[the remark this answers is not in the document]
Comment (bc-pi) [reply draft-ietf-oauth-selective-disclosure-jwt.md:1453]: "can be used" is correct in conveying that it... um... can be used.

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 8]
```diff
@@ -1407,6 +1413,41 @@ After the validation, the Verifier will have the following data for further proc
...
+
+The Disclosures of this SD-JWT are as follows:
+
+{{examples/pid/disclosures.md}}
+
+The Issuer added the following decoy digests:
+
+{{examples/pid/decoy_digests.md}}
+
+A presentation of the SD-JWT without a Key Binding JWT that discloses only biometrics based holder binding (`binding` of type `BiometricBinding`),
```
Comment (bc-pi) [inline draft-ietf-oauth-selective-disclosure-jwt.md:1441]:
```suggestion
A presentation of the SD-JWT without a Key Binding JWT that discloses only biometrics based holder binding (`binding` of type `BiometricBinding`),
```

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (Sakurann) [reply draft-ietf-oauth-selective-disclosure-jwt.md:1453]: agreed - it is only for those who wish to use this pubkey to validate the examples - we are not forcing anyone to do so :)

File: examples/w3c-vc/specification.yml [hunk 9]
```diff
@@ -1,7 +1,6 @@
 user_claims:
```
Comment (bc-pi) [inline examples/w3c-vc/specification.yml:1]: should this file be removed with this change to use arf-pid? also the one in w3c-vc_for_slide_deck?

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 10]
```diff
@@ -1343,47 +1349,54 @@ After the validation, the Verifier will have the following data for further proc
...
 <{{examples/complex_ekyc/verified_contents.json}}
 
-## Example 4a - Verifiable Credentials with JSON payload based on the SD-JWT format
+## Example 4a - SD-JWT-based Verifiable Credentials (SD-JWT VC)
 
-In this example, the artifacts defined in this specification are used to secure
-Verifiable Credentials with JSON payload based on the SD-JWT format as defined in [@I-D.terbu-sd-jwt-vc]. Key Binding is applied
-using the Holder's public key passed in a `cnf` Claim in the SD-JWT.
+In this example, the artifacts defined in this specification are used to represent
+Verifiable Credentials with JSON payload as defined in [@I-D.terbu-sd-jwt-vc].
```
Comment (bc-pi) [inline draft-ietf-oauth-selective-disclosure-jwt.md:1355]: maybe... ?

```suggestion
a verifiable credential with a JSON payload as defined in [@I-D.terbu-sd-jwt-vc].
```

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 11]
```diff
@@ -1343,47 +1349,54 @@ After the validation, the Verifier will have the following data for further proc
...
+{{examples/arf-pid/disclosures.md}}
 
-The following is how a presentation of the SD-JWT with Key Binding JWT that discloses only `given_name` and `is_over_18`
-claims with a Key Binding JWT could look like:
+The following decoy digests are added:
+
+{{examples/simple_structured/decoy_digests.md}}
 
-<{{examples/w3c-vc/sd_jwt_presentation.txt}}
+The following is how a presentation of the SD-JWT with Key Binding JWT that discloses only nationality and the fact that the person is over 18 years old could look like:
```
Comment (bc-pi) [inline draft-ietf-oauth-selective-disclosure-jwt.md:1381]:
```suggestion
The following is how a presentation of the SD-JWT with a Key Binding JWT that discloses only nationality and the fact that the person is over 18 years old could look like:
```

**Review by bc-pi — CHANGES_REQUESTED** [review]
a couple minor nits and maybe(?) some clean up of unused files but otherwise looks good

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 10 above)
Comment (bc-pi) [reply draft-ietf-oauth-selective-disclosure-jwt.md:1355]: or was that supposed to be the title of terbu-sd-jwt-vc? It read kinda funny with plurals or lack thereof. Also the title has changed.  So maybe... ? 
```suggestion
SD-JWT-based Verifiable Credentials (SD-JWT VC) as defined in [@I-D.terbu-sd-jwt-vc].
```

File: examples/w3c-vc/specification.yml (hunk 9 above)
Comment (bc-pi) [reply examples/w3c-vc/specification.yml:1]: I'd really like to remove unused stuff unless there's some compelling reason to keep it around.

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 10 above)
Comment (bc-pi) [reply draft-ietf-oauth-selective-disclosure-jwt.md:1355]: or title without the short part here: ... ?

 ```suggestion
SD-JWT-based Verifiable Credentials as defined in [@I-D.terbu-sd-jwt-vc].
```

File: examples/pid/specification.yml [hunk 12]
```diff
@@ -0,0 +1,42 @@
+user_claims:
+  iss: https://pid_issuer.memberstate.eu
```
[the remark this answers is not in the document]
Comment (peppelinux) [reply examples/pid/specification.yml]: memberstate.eu might collide with an existing fqdn

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 10 above)
Comment (Sakurann) [reply draft-ietf-oauth-selective-disclosure-jwt.md]: I like the middle option with the short part (yes, it was intended to be the doc title, thanks for catching!)

File: examples/w3c-vc/specification.yml (hunk 9 above)
Comment (Sakurann) [reply examples/w3c-vc/specification.yml:1]: I would like to have a place to have examples that are not referenced in the spec - you think danielfett/sd-jwt is a better option?

Comment (bc-pi) [reply examples/w3c-vc/specification.yml:1]: I'm not sure TBH but maybe. They aren't even examples but descriptor files that input into danielfett/sd-jwt tooling to generate examples. So living closer to danielfett/sd-jwt might be a better option. I was just looking to remove unused stuff here to keep this repo more 'clean' and make it easier to maintain. But we can leave it for now in order to move forward w/ getting this example in and publishing.

**Review by bc-pi — APPROVED** [review]

**Comment by bc-pi** [comment]
Thinking about this just a little more @Sakurann, does this warrant a doc history bullet?  Something like

"Replaced the general SD-JWT VC example with one based on Person Identification Data (PID) from the European Digital Identity Wallet Architecture and Reference Framework" 

The history is nice to have and I often use it as a strarting place when making "what's changed from last time" type slides for f2f meetings.

**Comment by bc-pi** [comment]
I'm just gonna add the doc history...

**Review by bc-pi — APPROVED** [review]
I think this is ready to merge (for real this time). It's kinda awkward process and numbers wise with Dr. out. And peppelinux is still listed as requesting changes but I think his comments were addressed. But I think it's good to go.

**Comment by bc-pi** [comment]
relevant section in the editor's copy preview: https://drafts.oauth.net/oauth-selective-disclosure-jwt/sakurann-pid-ex/draft-ietf-oauth-selective-disclosure-jwt.html#name-example-4a-sd-jwt-based-ver should anyone want to take a look before the coming soon merge

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 13]
```diff
@@ -553,7 +553,7 @@ established. For example, the Holder MAY provide a key pair to the Issuer,
 the Issuer MAY create the key pair for the Holder, or
 Holder and Issuer MAY use pre-established key material.
 
-Note: Examples in this document use the `cnf` Claim defined in [@RFC7800] to include
+Note: Examples in this document use the `cnf` claim defined in [@RFC7800] to include
```
Comment (peppelinux) [inline draft-ietf-oauth-selective-disclosure-jwt.md:556]:
```suggestion
Note: The examples in this document use the `cnf` claim defined in [@RFC7800] to include
```

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 14]
```diff
@@ -1262,6 +1238,12 @@ the media type is encoded as an SD-JWT.
   </front>
 </reference>
 
+<reference anchor="EUDIW.ARF" target="https://digital-strategy.ec.europa.eu/en/library/european-digital-identity-wallet-architecture-and-reference-framework">
```
Comment (peppelinux) [inline draft-ietf-oauth-selective-disclosure-jwt.md:1241]: Even if It May sounds paradoxal, this links Is not updated, I'll write today asking to update the arf version, since It Is still v1.0.0 while v1.2.0 Is going to be published

FYI below the fresh meat
https://github.com/eu-digital-identity-wallet/eudi-doc-architecture-and-reference-framework/release

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 15]
```diff
@@ -1319,7 +1301,7 @@ and `country` of the `address` property and without a Key Binding JWT could look
 
 ## Example 3 - Complex Structured SD-JWT {#example-complex-structured-sd-jwt}
 
-In this example, an SD-JWT with a complex object is demonstrated. Here, the data
+In this example, an SD-JWT with a complex object is demonstrated. The data
```
Comment (peppelinux) [inline draft-ietf-oauth-selective-disclosure-jwt.md:1304]:
```suggestion
In this example, an SD-JWT with a complex object is represented. The data
```

**Review by peppelinux — APPROVED** [review]
Approved with two pendings suggestioni/change request and a general comment that's not relevant for the revision but Just informative

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 14 above)
Comment (Sakurann) [reply draft-ietf-oauth-selective-disclosure-jwt.md:1241]: nice! thank you!
